components/storage/blockdevice/COMPONENT_OSPIF/OSPIFBlockDevice.h

```diff
@@ -0,0 +1,445 @@
+ *  #include "mbed.h"
+ *  #include "OSPIFBlockDevice.h"
+ *
+ *  OSPIFBlockDevice block_device(OSPI_FLASH1_IO0, OSPI_FLASH1_IO1, OSPI_FLASH1_IO2, OSPI_FLASH1_IO3, OSPI_FLASH1_IO4, OSPI_FLASH1_IO5, OSPI_FLASH1_IO6, OSPI_FLASH1_IO7
+ *                                OSPI_FLASH1_SCK, OSPI_FLASH1_CSN, OSPI_FLASH1_DQS, OSPIF_POLARITY_MODE_0, MBED_CONF_OSPIF_OSPI_FREQ);
+ *
+ *  int main()
+ *  {
+ *      printf("OSPI SFDP Flash Block Device example\n");
+ *
+ *      // Initialize the OSPI flash device and print the memory layout
+ *      block_device.init();
+ *      bd_size_t sector_size_at_address_0 = block_device.get_erase_size(0);
+ *
+ *      printf("OSPIF BD size: %llu\n",         block_device.size());
+ *      printf("OSPIF BD read size: %llu\n",    block_device.get_read_size());
+ *      printf("OSPIF BD program size: %llu\n", block_device.get_program_size());
+ *      printf("OSPIF BD erase size (at address 0): %llu\n", sector_size_at_address_0);
+ *
+ *      // Write "Hello World!" to the first block
+ *      char *buffer = (char *) malloc(sector_size_at_address_0);
+ *      sprintf(buffer, "Hello World!\n");
+ *      block_device.erase(0, sector_size_at_address_0);
+ *      block_device.program(buffer, 0, sector_size_at_address_0);
+ *
+ *      // Read back what was stored
+ *      block_device.read(buffer, 0, sector_size_at_address_0);
+ *      printf("%s", buffer);
+ *
+ *      // Deinitialize the device
+ *      block_device.deinit();
+ *  }
+ *  @endcode
+ */
+class OSPIFBlockDevice : public mbed::BlockDevice {
+public:
+    /** Create OSPIFBlockDevice - An SFDP based Flash Block Device over OSPI bus
+     *
+     *  @param io0 1st IO pin used for sending/receiving data during data phase of a transaction
+     *  @param io1 2nd IO pin used for sending/receiving data during data phase of a transaction
+     *  @param io2 3rd IO pin used for sending/receiving data during data phase of a transaction
+     *  @param io3 4th IO pin used for sending/receiving data during data phase of a transaction
+     *  @param io4 5th IO pin used for sending/receiving data during data phase of a transaction
+     *  @param io5 6th IO pin used for sending/receiving data during data phase of a transaction
+     *  @param io6 7th IO pin used for sending/receiving data during data phase of a transaction
+     *  @param io7 8th IO pin used for sending/receiving data during data phase of a transaction
+     *  @param sclk OSPI Clock pin
+     *  @param csel OSPI chip select pin
+     *  @param dqs OSPI dqs pin
+     *  @param clock_mode specifies the OSPI Clock Polarity mode (OSPIF_POLARITY_MODE_0/OSPIF_POLARITY_MODE_1)
+     *         default value = 0
+     *  @param freq Clock frequency of the OSPI bus (defaults to 40MHz)
+     */
+    OSPIFBlockDevice(PinName io0, PinName io1, PinName io2, PinName io3, PinName io4, PinName io5, PinName io6, PinName io7, PinName sclk, PinName csel, PinName dqs,
```

Comment:
I think you need to set default values as in QSPIFBlockDevice.h ?

Reply:
OK!
I have modified it.